Cesium.CodeGen/Contexts/TranslationUnitContext.cs

```diff
@@ -168,7 +168,8 @@ internal IType ResolveType(IType type)
         return type;
     }
 
-    internal TypeReference? GetTypeReference(IGeneratedType type) => _generatedTypes.GetValueOrDefault(type);
+    internal TypeReference? GetTypeReference(IGeneratedType type) => _generatedTypes.GetValueOrDefault(type) ??
+                                                                     _generatedTypes.FirstOrDefault(t => t.Key.Identifier == type.Identifier).Value;
```

Comment:
I don't believe this change is correct. The chance to mess something up (like `struct foo` vs `typedef foo struct { … }`) is very high.

I think that the type _should_ be emitted at this point.

For now, let's roll this statement back and fall back to `typedef` in a test, and figure the rest out in #453.

Reply:
Note I have rolled it back and pushed a commit to demonstrate an issue with this approach. Please tell if you are uncomfortable with such a move from me :)

Reply:
I knew something was wrong with this but it was the only way I found to get the type. I will work in #453 and we won't need this anymore.
No worries, I'm here to learn, any feedback is well appreciated.